File: Sorting/Insert_Sort.py
def InsertSort(arr):
    for i in range(1, len(arr)):
        if arr[i - 1] > arr[i]:
            temp = arr[i]
            j = i
            while(j > 0 and arr[j - 1] > temp):
                arr[j] = arr[j - 1]
                j -= 1
            arr[j] = temp


arr = [1, -2, 4, 7, 6, 3, 2, 3]
# arr = [3, 2, 3, 4, 6, 7]
InsertSort(arr)
print(arr)

File: Sorting/Insert_Sort.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    int a[] = {1,6,3,2,5,8,5,11};
    insert_sort(a,8);
    int i = 0;
    for(i = 0;i < 8; i++){
        printf("%d ", a[i]);
    }
    return 0;
}

void insert_sort(int s[], int len){
    int temp, i, j;
    for (i = 1; i < len; i++){
        if (s[i - 1] > s[i]){
            temp = s[i];
            j = i;
            while (j > 0 && s[j - 1] > temp)
            {
                s[j] = s[j - 1];
                j--;
            }
            s[j] = temp;
        }
    }
}
